src/Microsoft.AspNet.Mvc.Core/Filters/ProducesAttribute.cs

```diff
@@ -31,7 +32,13 @@ public override void OnResultExecuting([NotNull] ResultExecutingContext context)
 
             if (objectResult != null)
             {
-                SetContentTypes(objectResult.ContentTypes);
+                // Check if FormatFilter has already set the content type
```

Comment:
@harshgMSFT We're using the same logic here that API explorer uses, it takes into account scoping/ordering of filters.